src/memory.c

```diff
@@ -47,17 +47,17 @@ unsigned char dma_peek(long address)
     dmalist.count = 1;
     dmalist.source_addr = address & 0xffff;
     dmalist.source_bank = (address >> 16) & 0x0f;
-    dmalist.dest_addr = (unsigned int)&dma_byte;
+    dmalist.dest_addr = (uint16_t)&dma_byte;
     dmalist.dest_bank = 0;
 
     do_dma();
 
     return dma_byte;
 }
 
-unsigned char db1, db2, db3;
+uint8_t db1, db2, db3;
```

Comment:
These variables are used locally in the following function, why not put then inside it? Being outside like this their symbols have global visibility.
Look at the code, the do not even need to be static, it seems they can be ordinary auto variables?

Reply:
Makes sense to me and I don't know the original motivation. There might be different optimization strategies for cc65(?). Perhaps open a separate PR focusing on random?

Reply:
It may be because there is the idea that you should avoid local variables on some 6502 compilers, so perhaps it is best to do it conditionally depending on the compiler. It is probably best to leave it for now.